rosys/hardware/lizard_firmware.py

```diff
@@ -105,13 +105,20 @@ async def flash_core(self) -> None:
 
     async def flash_p0(self) -> None:
         rosys.notify(f'Flashing Lizard firmware {self.core_version} to P0...')
+        flash_p0_complete = False
+
+        def flash_callback():
+            nonlocal flash_p0_complete
+            flash_p0_complete = True
+        self.robot_brain.FLASH_P0_COMPLETE.register(flash_callback)
         await self.robot_brain.send('p0.flash()')
-        start = rosys.time()
-        while (self.robot_brain.hardware_time or 0) < start + 3.0:
-            if rosys.time() > start + 60.0:
+        timeout = rosys.time() + 120
```

Comment:
I usually distinguish between "timeout" as a duration and "deadline" as a point in time. Therefore I'd call `rosys.time() + 120` a deadline. But I'm not sure if this is common terminology.